toolchain/check/member_access.cpp

```diff
@@ -207,14 +228,38 @@ static auto PerformImplLookup(Context& context, Parse::NodeId node_id,
 // impl lookup if necessary. If the scope is invalid, assume an error has
 // already been diagnosed, and return BuiltinError.
 static auto LookupMemberNameInScope(Context& context, Parse::NodeId node_id,
-                                    SemIR::InstId /*base_id*/,
+                                    SemIR::InstId base_id,
                                     SemIR::NameId name_id,
                                     SemIR::ConstantId name_scope_const_id,
                                     LookupScope lookup_scope) -> SemIR::InstId {
   LookupResult result = {.specific_id = SemIR::SpecificId::Invalid,
                          .inst_id = SemIR::InstId::BuiltinError};
   if (lookup_scope.name_scope_id.is_valid()) {
-    result = context.LookupQualifiedName(node_id, name_id, lookup_scope);
+    AccessInfo access_info = {.base_id = base_id,
+                              .allow_private_access = false,
+                              .allow_protected_access = false};
+
+    if (auto class_type = context.insts().TryGetAs<SemIR::ClassType>(
+            context.constant_values().GetInstId(name_scope_const_id))) {
+      auto class_inst = context.classes().Get(class_type->class_id);
+
+      if (auto current_function = GetCurrentFunction(context)) {
+        auto is_access_from_instance_method = IsInstanceMethod(
+            context.sem_ir(), *current_function, class_inst.self_type_id);
+
+        if (is_access_from_instance_method) {
+          access_info.allow_private_access = true;
+          access_info.allow_protected_access = true;
+        }
+      }
+    }
+
+    result = context.LookupQualifiedName(node_id, name_id, lookup_scope,
+                                         /*required*/ true, access_info);
```

Comment:
```suggestion
                                         /*required=*/true, access_info);
```

The `=` here is significant for clang-format, and changes the spacing.